packages/grid/_modules_/grid/components/columnSelection/GridCellCheckboxRenderer.tsx

```diff
@@ -61,9 +62,12 @@ const GridCellCheckboxForwardRef = React.forwardRef<HTMLInputElement, GridCellPa
     }, [element, tabIndex]);
 
     React.useLayoutEffect(() => {
-      if (hasFocus && checkboxElement.current) {
-        const input = checkboxElement.current.querySelector('input')!;
-        input!.focus();
+      if (hasFocus) {
+        const input = checkboxElement.current?.querySelector('input');
+        input?.focus();
+      } else if (rippleRef.current) {
+        // Only available in @mui/material v5.4.1 or later
```

Comment:
https://github.com/mui-org/material-ui/pull/30901 is only available in this version.